packages/omgidl-serialization/src/MessageReader.test.ts

```diff
@@ -500,6 +513,57 @@ module builtin_interfaces {
     });
   });
 
+  it("reads multi-dimensional fixed size arrays", () => {
+    const msgDef = `
+        @mutable
+        struct Grid {
+          float table[2][3];
+        };
+    `;
+
+    const data = {
+      grid: [
+        [1, 2, 3],
+        [4, 5, 6],
+      ],
+    };
+    const writer = new CdrWriter({ size: 1028, kind: EncapsulationKind.PL_CDR2_LE });
+    writer.emHeader(true, 1, data.grid.length * data.grid[0]!.length * 4); // size of grid
+    for (const row of data.grid) {
+      writer.float32Array(row, false); // do not write length for fixed-size arrays
+    }
+
+    const rootDef = "Grid";
+    const reader = new MessageReader(rootDef, parseIdl(msgDef));
+    expect(reader.readMessage(writer.data)).toEqual({
+      table: [new Float32Array([1, 2, 3]), new Float32Array([4, 5, 6])],
+    });
+  });
+
+  it("PL_CDR2: reads an empty double (8-byte) array", () => {
```

Comment:
curious why you wrote this test case for PL_CDR2? Should we have equivalent tests for other kinds?

Reply:
Forgot to clean it up. I was also going to have PL_CDR1 support in this PR, but I pulled it out since it was small and needed the new CDR version